lib/rack/query_parser.rb

```diff
@@ -135,6 +134,18 @@ def params_hash_type?(obj)
       obj.kind_of?(@params_class)
     end
 
+    def params_hash_has_key?(hash, key)
+      return false if key =~ /\[\]/
```

Comment:
I'm probably missing something, but it would seem since you are splitting on `[]`, that the guard clause would ensure that `[]` exists in the query?

```ruby
return false if key !~ /\[\]/
```

Then again, I probably just missed the nuance here.

Yay make rack better!